komodo/symlink/sanity_check.py

```diff
@@ -0,0 +1,108 @@
+import os, sys
+import json
+import argparse
+import difflib
+import pprint
+
+
+def equal_links(a, b):
+    if a['root_folder'] != b['root_folder']:
+        return False
+
+    if set(a['root_links']) != set(b['root_links']):
+        return False
+
+    return a['links'] == b['links']
+
+
+def _linked_to(s_link, list_of_files):
+    
+    for file_name in list_of_files:
+        if os.path.islink(file_name):
+            if os.readlink(file_name) == s_link:
+                return True
+
+    return False
+
+
+def read_link_structure(path):
+    link_structure = {
+        'root_folder': os.path.realpath(path),
+        'root_links': [],
+        'links': {}
+    }
+
+    list_of_files = [os.path.join(path, file_name) for file_name in os.listdir(path)]
+    for file_path in list_of_files:
+        file_name = os.path.basename(file_path)
+        if os.path.islink(file_path):
+            link_structure['links'][file_name] = os.path.basename(os.readlink(file_path))
+            if not _linked_to(file_name, list_of_files):
+                link_structure['root_links'].append(file_name)
+
+    return link_structure
+
+
+def _check_link(link, link_dict, errors, visited=[]):
```

Comment:
Having mutable objects as default values are super scary. The recommended practice is something like:
```python
def (...., visited=None):
    if visited is None:
        visited = []
```

or to use an immutable structure:
```python
def (...., visited=()):
    # And then either:
    visited = list(visited)
    # or when adding
    visited += (link,)
```

Reply:
I think not having a default value is probably just as well. Was not aware of this issue..